data_node/upload_manager.go

```diff
@@ -102,8 +106,86 @@ func (um *UploadManager) handleInitialUpload(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	w.WriteHeader(http.StatusCreated)
 	w.Header().Set("ID", id)
+	w.Header().Set("Max-Request-Size", strconv.FormatInt(um.maxChunkSize, 10))
```

Comment:
Why not `fmt.Sprintf("%d", um.maxChunkSize)`? This will also cast to string and it will keep the base as is.